nemo_curator/utils/distributed_utils.py

```diff
@@ -271,6 +273,7 @@ def read_single_partition(
         add_filename: Whether to add a "filename" column to the DataFrame.
         input_meta: A dictionary or a string formatted as a dictionary, which outlines
             the field names and their respective data types within the JSONL input file.
+        columns: If not None, only these columns will be read from the file.
```

Comment:
It would be great if you could elaborate on how each of the file formats handles this parameter. Something simple like "Parquet files and cudf json files will get a performance bump from this."

Reply:
Updated, thanks. The comment (and PR) only includes Parquet gains for now, since there is some refactoring around handling JSON files that @praateekmahajan and I are planning to work on.